doc(README): Update docs to explain why you shouldn't set replicas in the deployment (#287)
DataDog/watermarkpodautoscaler: README.md

    Yes, WPA can scale on multiple metrics and works similar to [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#scaling-on-multiple-metrics). WPA evaluates each metric separately and proposes the number of replicas associated with the metric that requires the largest number. For example, if WPA evaluates metric1, metric2, metric3, and for each it calculates 10, 20, 30 replica proposals respectively, the final proposal is 30.

- **Should I specify replicas in my Deployment manifest?**

    No. If you define the replicas field in your Deployment spec, every time you apply the manifest, the replica count is reset to that specified value.
    For example, if you have replicas: 2 in your Deployment but the Watermark Pod Autoscaler (WPA) has scaled the Deployment up to 10 replicas, reapplying the manifest will revert the replica count to 2. This causes the Deployment to scale down immediately, and WPA must scale it back up again.

>     To avoid conflicts between the Deployment controller and WPA over the desired replica count, omit the replicas field entirely. This allows WPA to manage scaling without interference.

Reviewer: I think you can add a link to the official Kubernetes documentation about the technical reason: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#migrating-deployments-and-statefulsets-to-horizontal-autoscaling

    Yes, WPA can scale on multiple metrics and works similar to [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#scaling-on-multiple-metrics). WPA evaluates each metric separately and proposes the number of replicas associated with the metric that requires the largest number. For example, if WPA evaluates metric1, metric2, metric3, and for each it calculates 10, 20, 30 replica proposals respectively, the final proposal is 30.

> - **Should I set replicas on my deployment?**

Reviewer 2: Slightly rewriten version:

```
Should I specify replicas in my Deployment manifest?

No. If you define the replicas field in your Deployment spec, every time you apply the manifest, the replica count is reset to that specified value.
For example, if you have replicas: 2 in your Deployment but the Watermark Pod Autoscaler (WPA) has scaled the Deployment up to 10 replicas, reapplying the manifest will revert the replica count to 2. This causes the Deployment to scale down immediately, and WPA must scale it back up again.
To avoid conflicts between the Deployment controller and WPA over the desired replica count, omit the replicas field entirely. This allows WPA to manage scaling without interference.
```

Author: Updated to reflect this wording, I agree this is clearer phrasing.

    Yes, WPA can scale on multiple metrics and works similar to [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#scaling-on-multiple-metrics). WPA evaluates each metric separately and proposes the number of replicas associated with the metric that requires the largest number. For example, if WPA evaluates metric1, metric2, metric3, and for each it calculates 10, 20, 30 replica proposals respectively, the final proposal is 30.

- **Should I set replicas on my deployment?**

>     No. If you set the `replicas` value on your deployment, then every time you apply the deployment, the number of replicas will be reset to that value.

Reviewer 3: ```suggestion
    No. If you set the `replicas` value on your deployment, then every time you apply the deployment, the number of replicas are reset to that value.
```

    This means that if you have `replicas: 2` specified in your deployment but WPA has scaled up the deployment to be 10 pods, then when you apply the 

>     deployment the number of pods will quickly become 2 and you'll have to wait until WPA scales up again. 

Reviewer 3: ```suggestion
    deployment the number of pods quickly become 2 and you have to wait until WPA scales up again. 
```

>     To avoid this, omit setting `replicas` in your deployment. 

Reviewer 3: ```suggestion
```

>     This ensures that there's no races between Kubernetes and WPA in setting the number of replicas correctly.

Reviewer 3: ```suggestion
    Omitting `replications` from your deployment ensures that there are no races between Kubernetes and WPA to set the number of replicas correctly.
```

Author: I can change this to "Omitting `replicas` from your deployment ..." but replications seems like its incorrect.
